demos/PointAdapter/super4pcs_adapter_test.cc

```diff
@@ -0,0 +1,258 @@
+  vector<typename Point3D::VectorType> normals1, normals2;
+  vector<tripple> tris1, tris2;
+  vector<std::string> mtls1, mtls2;
+
+  // Match and return the score (estimated overlap or the LCP).
+  typename Point3D::Scalar score = 0;
+
+  constexpr Utils::LogLevel loglvl = Utils::Verbose;
+  using SamplerType   = gr::UniformDistSampler;
+  using TrVisitorType = typename std::conditional <loglvl==Utils::NoLog,
+                            DummyTransformVisitor,
+                            TransformVisitor>::type;
+  using PairFilter = gr::AdaptivePointFilter;
+
+  SamplerType sampler;
+  TrVisitorType visitor;
+  Utils::Logger logger(loglvl);
+
+  /// TODO Add proper error codes
+  if(argc < 4){
+      Demo::printUsage(argc, argv);
+      exit(-2);
+  }
+  if(int c = Demo::getArgs(argc, argv) != 0)
+  {
+    Demo::printUsage(argc, argv);
+    printS4PCSParameterList();
+    exit(std::max(c,0));
+  }
+
+  // prepare matcher ressourcesoutputSampled2
+  using MatrixType = Eigen::Matrix<typename Point3D::Scalar, 4, 4>;
+  MatrixType mat (MatrixType::Identity());
+
+  // Read the inputs.
+  if (!iomananger.ReadObject((char *)input1.c_str(), set1, tex_coords1, normals1, tris1,
+                  mtls1)) {
+    logger.Log<Utils::ErrorReport>("Can't read input set1");
+    exit(-1);
+  }
+
+  if (!iomananger.ReadObject((char *)input2.c_str(), set2, tex_coords2, normals2, tris2,
+                  mtls2)) {
+    logger.Log<Utils::ErrorReport>("Can't read input set2");
+    exit(-1);
+  }
+
+  // clean only when we have pset to avoid wrong face to point indexation
+  if (tris1.size() == 0)
+    Utils::CleanInvalidNormals(set1, normals1);
+  if (tris2.size() == 0)
+    Utils::CleanInvalidNormals(set2, normals2);
+
+  //
+  // Convert the points to PointType1 to test adapter
+  //
+  std::vector<extlib1::PointType1> set1_points1 = getExtlib1Points(set1);
+  std::vector<extlib1::PointType1> set2_points1 = getExtlib1Points(set2);
+
+  try {
+
+      if (use_super4pcs) {
+          using MatcherType = gr::Match4pcsBase<gr::FunctorSuper4PCS, extlib1::PointAdapter, TrVisitorType, gr::AdaptivePointFilter, gr::AdaptivePointFilter::Options>;
+          using OptionType  = typename MatcherType::OptionsType;
+
+          OptionType options;
+          if(! Demo::setOptionsFromArgs(options, logger))
+          {
+            exit(-2); /// \FIXME use status codes for error reporting
+          }
+/* template<class PointType, class Range, class Matcher, class Options, class Sampler, class TransformVisitor> typename PointType::Scalar computeAlignment(const Options&, const gr::Utils::Logger&, const Range&, const Range&, Eigen::Ref<Eigen::Matrix<typename PointType::Scalar, 4, 4> >, const Sampler&, TransformVisitor&) */
+          extlib1::PointAdapter::Scalar score = computeAlignment<extlib1::PointAdapter, std::vector<extlib1::PointType1>, MatcherType, OptionType, SamplerType, TrVisitorType> (options, logger, set1_points1, set2_points1, mat, sampler, visitor);
```

Comment:
You don't need to specify the template types if they can be deduced by the compiler: `Range`, `Options`, `Sampler` and `TransformVisitor`.
You may also need to reorder the template parameter to have all the deduced types on the right side of the others.

Reply:
solved: https://github.com/necipfazil/OpenGR/commit/66bfc76354bc98837516d6460d6ab6ce30cc81e4